### ISSUE TITLE
normal_transform_pixel3d and normalize_homography3d have no align_corners, so warp_affine3d(align_corners=False) is wrong (3D half of #3904)

### ISSUE THREAD

**Issue by VanniLeonardo** [kornia/kornia#4503]
### What happened?

This is the 3D half of #3904, which #3945 fixes for 2D.

`normal_transform_pixel3d` and `normalize_homography3d` hardcode the corner-aligned normalization, where voxel centres `0` and `size - 1` map to `-1` and `+1`. Neither takes an `align_corners` argument. `warp_affine3d` then hands the matrix built that way to `grid_sample(align_corners=...)`, which reads `-1` and `+1` as the outer voxel edges when the flag is `False`. Mixing the two conventions applies a spurious sub-pixel scale and shift.

A whole-voxel translation is the clearest case, because it should move the volume by exactly one voxel under either convention. At `align_corners=False` it does not.

This is the same shape of defect as the 2D one, and the 2D figure on current `main` is in the reproduction below for comparison. #3945 threads `align_corners` through `create_meshgrid`, `normal_transform_pixel`, `normalize_homography` and `denormalize_homography`, but deliberately leaves the 3D functions alone, because the 3D test expectations need their own review.

### Minimal reproduction

```python
import inspect, torch
from kornia.geometry.conversions import normal_transform_pixel, normal_transform_pixel3d, normalize_homography3d
from kornia.geometry.transform import warp_affine3d, warp_affine

print("2D normal_transform_pixel  :", inspect.signature(normal_transform_pixel))
print("3D normal_transform_pixel3d:", inspect.signature(normal_transform_pixel3d))
print("3D normalize_homography3d  :", inspect.signature(normalize_homography3d))

d, h, w = 2, 3, 4
vol = torch.arange(float(d * h * w)).view(1, 1, d, h, w)
M = torch.eye(4)[None][:, :3].clone()
M[:, 0, 3] = 1.0                                   # shift by +1 voxel along x
expected = torch.zeros_like(vol)
expected[..., 1:] = vol[..., :-1]
for ac in (True, False):
    out = warp_affine3d(vol, M, (d, h, w), align_corners=ac)
    print(f"warp_affine3d  +1 voxel, align_corners={ac!s:5s} max|out - expected| = {(out - expected).abs().max():.6f}")

img = torch.arange(12.0).view(1, 1, 3, 4)
M2 = torch.tensor([[[1.0, 0.0, 1.0], [0.0, 1.0, 0.0]]])
exp2 = torch.zeros_like(img)
exp2[..., 1:] = img[..., :-1]
for ac in (True, False):
    out = warp_affine(img, M2, (3, 4), align_corners=ac)
    print(f"warp_affine 2D +1 pixel, align_corners={ac!s:5s} max|out - expected| = {(out - exp2).abs().max():.6f}")
```

Observed on current `main`:

```
2D normal_transform_pixel  : (height, width, eps=1e-14, device=None, dtype=None)
3D normal_transform_pixel3d: (depth, height, width, eps=1e-14, device=None, dtype=None)
3D normalize_homography3d  : (dst_pix_trans_src_pix, dsize_src, dsize_dst)

warp_affine3d  +1 voxel, align_corners=True  max|out - expected| = 0.000000
warp_affine3d  +1 voxel, align_corners=False max|out - expected| = 6.666670
warp_affine 2D +1 pixel, align_corners=True  max|out - expected| = 0.000000
warp_affine 2D +1 pixel, align_corners=False max|out - expected| = 2.666668
```

The 2D row is what #3945 takes to `0.000000`. The 3D row is what this issue is about.

Expected: `align_corners` selects only how coordinates are normalized for `grid_sample`, never the geometry. A whole-voxel translation moves the volume by exactly one voxel under both settings, and the two agree except in out-of-bounds handling.

### Environment

- Kornia version: 0.9.0rc1 (current `main`)
- PyTorch version: 2.9.1+cu128
- Python version: 3.11.14
- OS / device: Linux, CPU

### Anything else?

Suggested fix, mirroring #3945:

- give `normal_transform_pixel3d`, `normalize_homography3d` and any 3D counterpart of `denormalize_homography` an `align_corners: bool = True` argument, last in the signature, so existing calls stay bit-identical;
- use the half-pixel mapping `x_norm = (2x + 1) / size - 1` when it is `False`, which needs no singleton guard because a size of 1 already lands on the centre;
- thread the flag from `warp_affine3d` and `warp_perspective3d` down into the normalization they build;
- review the 3D test expectations that change, keeping the pre-fix literal next to each new one, the way #3945 does.

Worth doing after #3945 lands so the two stay consistent. I am happy to take this one.

Note that `warp_perspective3d` cannot be verified by this route until the separate `create_meshgrid3d` channel-order defect is fixed, filed as #4502. That one breaks the 3D perspective path under both `align_corners` settings.

Found while preparing the 3D follow-up requested in the #3945 review. Investigated with Claude Code; every figure above is from an executed run on current `main`.

**Comment by ducha-aiki** [comment]
Reopening: this was closed by accident, and the defect is still on `main`.

- **Why it closed:** #4786, a docs-only change, was merged at 14:29:06. #4503 closed one second later, with no fixing commit. The #4786 body contains "…when it fixes #4503", where the sentence is about #4556. GitHub's closing-keyword parser does not read the subject of the sentence, so GraphQL `closingIssuesReferences` lists #4503 for #4786, and for #4788, which repeats the phrase. Neither PR touches the 3D normalization.
- **Still present on `main` (750c1611e, CPU, float32/float64):**
  - `test_wart_random_perspective3d_identity_align_corners_false_4503` passes.
  - The strict xfails `test_convention_random_perspective3d_identity_align_corners_false` and `TestWarpPerspective3d::test_convention_identity_reproduces_input_align_corners_false` still xfail.
- **The fix is still pending:** #4556 is open and lists #4503 as its closing reference, so this closes properly when that PR merges.

Posted on behalf of @ducha-aiki by Claude (Opus 5.5).
